.gitignore

```diff
@@ -75,8 +75,10 @@ m4/
 .libs/
 *.lo
 *.la
-tests/integration/cppclient.t
+/tests/integration/cppclient.t
 libtool
 ltmain.sh
 cppclient/watchmanclient.pc
 /website/.jekyll-metadata
+/rust/**/Cargo.lock
```

Comment:
Not sure if this is intended to build reproducibly, but: it's a good idea to commit `Cargo.lock` files.

Happy to see this!

Reply:
The general rule is to commit Cargo.lock files for binaries but not for libraries.

Reply:
Gotcha. But presumably there is a "root" somewhere in here that you're planning to load? A `cdylib` or something? I believe that you'd want a reproducible build starting from whatever your root is.

Reply:
If I understood you correctly, you're asking for what happens when we decide to link the Rust bindings into the Watchman binary. (Please correct me if I'm wrong! :) )

We're not planning to load the Rust bindings into the main Watchman binary at all, actually. The Rust bindings are going to be provided for Rust projects to use to talk to Watchman out of process. I agree that if and when we decide to load the Rust bindings into the main Watchman binary, we'd want a Cargo.lock.

Hope that helps :)

Reply:
Ahh, I see. Yep, that helps. Thanks!